 # SACRAMENTO_Laboratory Activity 2

## PROBLEM #1

1. In an empty cell, declare a variable value and assign it the value of 5 then display its value using print().

In [1]:
value = 5
print(value)

5


2. Create a new cell and type the command: type(value) then run the cell. The output should be like the image below.

In [2]:
type(value)

int

3. In a new cell, use the same variable value and assign it the value of 5.0 then print the value. 

In [3]:
value = 5
print(value)

5


4. Repeat step 2. 
Note: You may choose to decide how you execute the code in the cells for the next tasks in the procedure.

5. Repeat these steps for the following values:
a. 2+3j

b. ‘Hello World’

c. “Hello World”

d. True

e. False

f. [1,2,3,4,5]

g. (1,2,3,4,5)

h. { ‘name’: ‘Your_name’ }

i. None

In [4]:
type(2+3j)

complex

In [5]:
type('Hello World')

str

In [6]:
type("Hello World")

str

In [7]:
type(True)

bool

In [8]:
type(False)

bool

In [9]:
type([1,2,3,4,5,])

list

In [10]:
type((1,2,3,4,5))

tuple

In [11]:

type({ 'name': 'Your_name' })

dict

In [12]:
type(None)

NoneType

6. Re-assign the value variable to be equal to 5.

In [13]:
value = 5

7. Declare a new variable named value2 to be equal to -6.

In [14]:
value2 = -6

## PROBLEM #2


Performing Operations with Python
1. Using value and value2. Type the command: print(value+value2)

In [15]:
print(value+value2)
print(value-value2)


-1
11


2. Repeat step 1 for the following values of value and value2:

In [16]:
value, value2 = 5.0,6
print(value+value2)
print(value-value2)

11.0
-1.0


In [17]:
value, value2 = -5 , 6.1
print(value+value2)
print(value-value2)

1.0999999999999996
-11.1


In [18]:
value, value2 = "Hello" , ' world'
print(value+value2)
print(value-value2)

Hello world


TypeError: unsupported operand type(s) for -: 'str' and 'str'

In [19]:
value, value2 = [1,2,3],[4,5,6]
print(value+value2)
print(value-value2)

[1, 2, 3, 4, 5, 6]


TypeError: unsupported operand type(s) for -: 'list' and 'list'

In [20]:
value, value2 = (1,2,3),(4,5,6)
print(value+value2)
print(value-value2)

(1, 2, 3, 4, 5, 6)


TypeError: unsupported operand type(s) for -: 'tuple' and 'tuple'

In [21]:
value, value2 =  {"name":"Royce"}, {"age":2}
print(value + value2)
print(value-value2)


TypeError: unsupported operand type(s) for +: 'dict' and 'dict'

3. Using value, value2 = 30, 4. Type the commands: 

In [22]:
value , value2 = 30,4
print(value*value2)


120


In [23]:
print(value2**2)

16


In [24]:
print(value2**3)

64


In [25]:
print(value*value2+value2**2+1)

137


In [26]:
print(value/value2)

7.5


In [27]:
print(value%value2)

2


## PROBLEM #3

Assigning Input Data to a Variable
Finding a person’s BMI (metric)
1. Declare a new variable named name and assign it the value input(“Enter your name”)

In [9]:
name = (input("Enter your name:"))

Enter your name:John Ray


2. Create another variable named weight and assign it the value input(“Enter your weight(kg): “)

In [2]:
weight = (input("Enter your weight(kg):"))

Enter your weight(kg):55


3. Create another variable named height and assign it the value input(“Enter your meters(m): “)

In [3]:
height = (input("Enter your meters(m): "))

Enter your meters(m): 2


4. Declare another variable called bmi and assign it the formula  

In [4]:
bmi = weight / (height*height)

TypeError: can't multiply sequence by non-int of type 'str'

5. Address the errors displayed step#4. You can accomplish this by converting the String input to another data type. An example would be:

weight = input(“Enter your weight(kg)”)
weight = float(weight)

Or simply weight = float(input(“Enter your weight(kg): “))

There are many functions available that can convert one data type to another. Some of which are the following:
int(), float(), str()

Other functions which maybe used in the later lab activities are: complex(real, imaginary), list(), tuple(), set(), dict(), ord(), bin(), hex(), oct(). 

In [8]:

weight = float (input("Enter your weight(kg):"))

Enter your weight(kg):60


In [10]:
height =  float (input("Enter your meters(m): "))

Enter your meters(m): 1.6764


In [13]:
bmi = weight / (height**2)

6. Print the persons’s name, weight, height, and bmi
Name: John Ray
Weight: 60
Height:  1.6764 
BMI = 21.3499

In [14]:
print("Name:",name)
print("Weight :",weight)
print("Height: ", height)
print("BMI = ",bmi)

Name: John Ray
Weight : 60.0
Height:  1.6764
BMI =  21.349904958763087
